### PR TITLE
Add more tests and examples

### PR DESCRIPTION
**Pull request by davidkonigsberg**
- Added trade search example
- Adding tests to match python and go SDKs

### CODE REVIEW

File: examples/trade-search.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import * as dotenv from "dotenv";
+import { SayariClient } from "../dist/index.js";
+
+// Pull Client creds from ENV file
+dotenv.config({ path: '../.env' });
+dotenv.config();
+var clientID = process.env.CLIENT_ID;
```
Comment (mtomran) [inline examples/trade-search.mjs:7]: better to use const or let instead of var

File: examples/trade-search.mjs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+var clientSecret = process.env.CLIENT_SECRET;
+var baseURL = 'https://api.sayari.com';
+
+// use BASE_URL if provided
+if (process.env.BASE_URL) {
+    baseURL = String(process.env.BASE_URL)
+}
```
Comment (mtomran) [inline examples/trade-search.mjs:14]:
```suggestion
const baseURL = process.env.BASE_URL ||  'https://api.sayari.com'
```

File: examples/trade-search.mjs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+// Create an authenticated sayari client
+const client = new SayariClient({ clientId: clientID, clientSecret: clientSecret, environment: baseURL })
+
+// Search for shipments
+const shipments = await client.trade.searchShipments({q:"microcenter"})
+console.log("Found", shipments.data.length, "shipments.")
```
Comment (mtomran) [inline examples/trade-search.mjs:21]: nit:
```suggestion
console.log(`Found ${shipments.data.length} shipments.`)
```

File: tests/sdk.test.ts [hunk 4]
```diff
@@ -0,0 +1,286 @@
+import { SayariClient } from "../src/index";
+
+var clientID = String(process.env.CLIENT_ID);
```
Comment (mtomran) [inline tests/sdk.test.ts:3]: I don't think you need to call `String()` since env vars are already sting

File: tests/sdk.test.ts [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+var clientSecret = String(process.env.CLIENT_SECRET);
+var baseURL = 'https://api.sayari.com';
+
+// use BASE_URL if provided
+if (process.env.BASE_URL) {
+    baseURL = String(process.env.BASE_URL)
+}
```
Comment (mtomran) [inline tests/sdk.test.ts:10]: same as above

File: tests/sdk.test.ts [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+
+            // Get entity details
+            const entityDetails = await client.entity.getEntity(firstEntity.id);
+            expect(entityDetails.label).toEqual(firstEntity.label)
+            expect(entityDetails.degree).toEqual(firstEntity.degree)
+            expect(entityDetails.pep).toEqual(firstEntity.pep)
+            expect(entityDetails.psaCount).toEqual(firstEntity.psaCount)
+            expect(entityDetails.type).toEqual(firstEntity.type)
+
+            done = true
```
Comment (mtomran) [inline tests/sdk.test.ts:69]: can use `break` here to break out of the loop

File: tests/sdk.test.ts [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+            const randString = generateRandomString(3)
+
+            const resolution = await client.resolution.resolution({name: randString});
+            if (resolution.data.length == 0) {
+                continue
+            }
+            expect(resolution.data.length).toBeGreaterThan(0)
+            expect(resolution.fields.name).toEqual([randString])
+
+            done = true
```
Comment (mtomran) [inline tests/sdk.test.ts:86]: same as above, can use `while(true)` and `break`
also in other tests

File: tests/sdk.test.ts [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+        let done = false;
+        while (!done) {
+            const randString = generateRandomString(3)
+
+            // get an entity
+            const entitySearchResults = await client.search.searchEntity({q: randString});
+            if (entitySearchResults.data.length == 0) {
+                console.log("no result, will retry with a different search term")
+                continue
+            }
```
Comment (mtomran) [inline tests/sdk.test.ts:138]: This section seems to be repeated, so could move to a function like 
```
async function runAndRetry(func) {
  while(true) {
    const randString = generateRandomString(3)
    result = await func({q: randString})
    if (result.data.length === 0) {
      console.log("no result, will retry with a different search term")
      continue
    }
    break
  }
  return result
}
```
and call 
```
const entitySearchResults = await runAndRetry(client.search.searchEntity)
```

**Review by mtomran — APPROVED** [review]
Looks Good! only a few suggestions

File: tests/sdk.test.ts (hunk 4 above)
Comment (davidkonigsberg) [reply tests/sdk.test.ts:3]: I tried without and I get an error:
```
    tests/sdk.test.ts:13:5 - error TS2322: Type 'string | undefined' is not assignable to type 'Supplier<string>'.
      Type 'undefined' is not assignable to type 'Supplier<string>'.

    13     clientId: clientID,
           ~~~~~~~~

      src/Client.ts:25:9
        25         clientId: core.Supplier<string>;
                   ~~~~~~~~
        The expected type comes from property 'clientId' which is declared here on type 'Options'

```

File: tests/sdk.test.ts (hunk 8 above)
Comment (davidkonigsberg) [reply tests/sdk.test.ts:138]: I have tried a bunch of approaches to this without any luck. The closest I can was this, but when I called it I got complaints about properties on the client being undefined....

File: tests/sdk.test.ts (hunk 4 above)
Comment (mtomran) [reply tests/sdk.test.ts]: You have to take care of undefined case. Either `const clientId = process.env.CLIENT_ID || ''` or deal with it differently maybe throw an Error?

**Pull request merged**